### PR TITLE
feat: added Rules Page, Navbar and Footer

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Added Rules page, Navbar and Footer

# More details:

1. Rules Page -
- On entering this page, the user is required to enter her/his name.
- After submitting the name, the user will see the rules. 

2. Navbar -
a. Logo
b. Search icon (not working)
c. Favorites
d. Login button
e. Sign up button

3. Footer - 
- Links to my social media pages/ contact information.

4. Responsive

# How to access rules page - 
Click on `Play now` button.

# Screenshots:

RulesName
rules

### CODE REVIEW

File: src/Components/Footer/Footer.css [hunk 1]
```diff
@@ -1,6 +1,22 @@
 .siteInfo {
   width: 98.9vw;
+  background-color: black;
+  color: aliceblue;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
   height: 8vh;
+  padding-top: 4px;
+  border-top: 1px solid blue;
```
Comment (ranitasahaa1230) [inline src/Components/Footer/Footer.css]: Put all these colors in your root variables

File: src/Components/Footer/Footer.css [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+}
+
+.contact {
+  color: white;
   background-color: black;
+}
+
+.footer--icons i {
   color: white;
```
Comment (ranitasahaa1230) [inline src/Components/Footer/Footer.css]: Same here

**Review by ranitasahaa1230 — APPROVED** [review]
Code looks clean
Just a suggestion:- In the UI, when the user is not entering their name/skipping that part, and on the click of a submit button, the page is still redirecting to the rules page. Why don't you put the name field as mandatory or you can also show an error message like "Please enter all fields"!

File: src/Components/Footer/Footer.css (hunk 2 above)
Comment (krgarima) [reply src/Components/Footer/Footer.css]: done

File: src/Components/Footer/Footer.css (hunk 1 above)
Comment (krgarima) [reply src/Components/Footer/Footer.css]: Disabled the button until user enters his/her name. Thanks.

Comment (krgarima) [reply src/Components/Footer/Footer.css]: Added CSS variables.

File: src/index.js [hunk 3]
```diff
@@ -10,12 +10,15 @@ import Result from "./Pages/Result/Result";
 import Rules from "./Pages/Rules/Rules";
 import NavBar from "./Components/Navbar/Navbar";
 import Footer from "./Components/Footer/Footer";
+import { NameContextProvider } from "./Context/name-context";
 
 ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
       <CategoryContextProvider>
```
Comment (sashaboi) [inline src/index.js:18]: are we yet to use category context ? I see that the import might be pending

File: src/Pages/Rules/Rules.js [hunk 4]
```diff
@@ -1,5 +1,60 @@
...
+          <input
+            type="text"
+            id="userName"
+            onChange={(e) => {
+              setName(e.target.value);
+            }}
+          />
+
+          <button
+            disabled={name.length === 0}
```
Comment (sashaboi) [inline src/Pages/Rules/Rules.js:44]: nice touch, very cool!

**Review by sashaboi — APPROVED** [review]
I found only one thing which might be changed. 

Other than that here's the good practices followed which I inspire to do myself as well!
CSS naming very clean.
button disabled very cool. 
overall folder structure , readability , really nice! 

Do take care about the one comment and good to go!
Onkar Deshpande

File: src/index.js (hunk 3 above)
Comment (krgarima) [reply src/index.js:18]: No, the import is not pending. All these files are being exported to the Home page where we need category to render cards of a particular category.

File: src/Pages/Rules/Rules.js (hunk 4 above)
Comment (krgarima) [reply src/Pages/Rules/Rules.js:44]: Thank you

**Pull request merged**